### ISSUE TITLE
Implement numba in cdms

### ISSUE THREAD

**Issue by durack1** [CDAT/cdms#79]
@dnadeau4 this link (discussing cython vs numba and speedups using each) was interesting.. https://medium.com/@elgehelge/the-5-most-important-python-data-science-advancements-of-2015-a136482da89b#.4psi1vuqy
![](https://cdn-images-1.medium.com/max/800/1*FPTxvMefHjaoA8a5pdGoZw.png)

**Comment by doutriaux1** [comment]
@durack1 it's on @dnadeau4 list we have a proposal in for this, we are also looking at other tools such as [DASK](http://dask.pydata.org/en/latest/) etc...

**Comment by durack1** [comment]
@doutriaux1 yep, I discussed this with @dnadeau4 yesterday the above link was wedged in another issue which was more relevant to numba so it was copied across above and the other issue (upgrade numpy to 1.10 was closed)

**Comment by zshaheen** [comment]
@durack1 @doutriaux1 @dnadeau4 I was actually looking into Cython for some metrics stuff I hope to do soon. numba is cool, but do note it's limited/specialized for numpy code.

I just read [this](https://www.amazon.com/High-Performance-Python-Performant-Programming/dp/1449361595/ref=sr_1_1?ie=UTF8&qid=1484939791&sr=8-1&keywords=high+performance+python) book and it explains the different ways to speed up python code. It's really good.

**Comment by doutriaux1** [comment]
@zshaheen I don't really care for Cython. I don't think it really applies to us. We usually control the `C` code so I think it's better to simply embed C into our Python directly. Cython if I recall correctly is mostly useful if you get canned `C` libraries that you do not necessarily control.
